### PR TITLE
Adding ability for temp controller to be disconnected.

### PR DESCRIPTION
**Pull request by JosephParsons**

### CODE REVIEW

**Comment by JosephParsons** [comment]
Parker has confirmed it works in the lab.

File: python/lsst/ts/tunablelaser/component.py [hunk 1]
```diff
@@ -515,31 +516,60 @@ def __init__(
             encoding=encoding,
             simulation_mode=simulation_mode,
         )
-        self.e5dc_b = canbus_modules.E5DCB(
-            component=self, simulation_mode=simulation_mode
-        )
         self.lock = asyncio.Lock()
 
-        self.host = host
+        # if host is not valid IP address assume its unconnected
```
Comment (couger01) [inline python/lsst/ts/tunablelaser/component.py:521]: Do we want to restrict to only IP addresses and not have hostname/DNS support?

File: python/lsst/ts/tunablelaser/component.py [hunk 2]
```diff
@@ -515,31 +516,60 @@ def __init__(
...
+            ip_address(host)
+            self.host = host
+            self.e5dc_b = canbus_modules.E5DCB(
+                component=self, simulation_mode=simulation_mode
+            )
+        except ValueError:
+            self.log.error(
+                f"Host address given to Temp Ctrl not valid, assuming unconnected: {host}"
+            )
+            self.host = "unconnected"
```
Comment (couger01) [inline python/lsst/ts/tunablelaser/component.py:532]: I think self.host should be None.

File: python/lsst/ts/tunablelaser/interfaces.py [hunk 3]
```diff
@@ -227,11 +227,12 @@ async def connect(self):
...
             self.port = 50
-        self.commander = tcpip.Client(
-            host=self.host,
-            port=self.port,
-            log=self.log,
-            terminator=bytes(self.terminator),
-            encoding=self.encoding,
-        )
-        await self.commander.start_task
+        if self.host != "unconnected":
```
Comment (couger01) [inline python/lsst/ts/tunablelaser/interfaces.py:230]: This should be `self.host is not None`

File: tests/data/config/disconnected_temp_ctrl.yaml
Comment (couger01) [inline tests/data/config/disconnected_temp_ctrl.yaml:1]: Add a newline.

**Review by couger01 — CHANGES_REQUESTED** [review]
Had a few suggestions.

**Comment by parfa30** [comment]
Is this ready to go?

File: python/lsst/ts/tunablelaser/interfaces.py [hunk 4]
```diff
@@ -224,11 +224,16 @@ async def disconnect(self):
...
         """Connect to the module."""
-        self.commander = tcpip.Client(
-            host=self.host,
-            port=self.port,
-            log=self.log,
-            terminator=bytes(self.terminator),
-            encoding=self.encoding,
-        )
-        await self.commander.start_task
+        # TODO remove if jenkins pass
```
Comment (couger01) [inline python/lsst/ts/tunablelaser/interfaces.py:227]: Remove this todo or add a ticket to it.

File: python/lsst/ts/tunablelaser/mock_server.py [hunk 5]
```diff
@@ -113,21 +114,39 @@ def __init__(self, host=tcpip.LOCAL_HOST, port=0) -> None:
...
+                encoding="ascii",
+            )
+        except ValueError:
+            self.log.error(
+                f"TempCtrlServer hostname was invalid, assuming temp ctrler unconnected: {host}"
+            )
+            super().__init__(
+                name="TempCtrl Mock Server",
+                host=tcpip.LOCAL_HOST,
+                port=50,
```
Comment (couger01) [inline python/lsst/ts/tunablelaser/mock_server.py:134]: Don't default to 50 as that is a privileged port and can cause issues with the CI.

**Review by couger01 — APPROVED** [review]
LGTM.

**Pull request merged**